src/Testing/ErrorFormatterTestCase.php

```diff
@@ -78,6 +78,7 @@ protected function getAnalysisResult(int $numFileErrors, int $numGenericErrors):
 			throw new ShouldNotHappenException();
 		}
 
+		/** @var list<Error> */
```

Comment:
why are these phpdoc changes required? is it related to your extension changes?

Reply:
> why are these phpdoc changes required? is it related to your extension changes?

yes, it's related to the changes. the `array_slice` is invoked with `$numFileErrors` as the 3rd (limit) param, so with the new changes it will return `array` as type not a dynamic type as before (in this case `list<Error>`) so I had to add phpdoc to override this.

this is safe because here https://github.com/phpstan/phpstan-src/blob/624edbbd84f346bab195bbfebcf6e5993a180bb1/src/Testing/ErrorFormatterTestCase.php#L77 is a check for range of `$numFileErrors` so array_slice will always return a list<Error> considering input is a constant array, but in general `array_slice` shouldn't do dynamic types if offset or limit are variables as output cannot be predicted, hence this PR.

Reply:
> will always return a list

The extension should reflect that. With your changes the function is now less usable than before. We shouldn't need `/** @var list<Error> */` here.